packages/shared/src/react/hooks/useReverification.ts

```diff
@@ -74,16 +103,24 @@ function createReverificationHandler(params: { onOpenModal: Clerk['__experimenta
   return assertReverification;
 }
 
-function __experimental_useReverification<Fetcher extends () => Promise<any>>(fetcher: Fetcher): readonly [Fetcher] {
+function __experimental_useReverification<
+  Fetcher extends (...args: any[]) => Promise<any>,
+  O extends UseReverificationOptions,
+>(
+  fetcher: Fetcher,
+  options?: O,
+): readonly [(...args: Parameters<Fetcher>) => Promise<ExcludeClerkError<Awaited<ReturnType<Fetcher>>, O>>] {
   const { __experimental_openUserVerification } = useClerk();
   const fetcherRef = useRef(fetcher);
+  const optionsRef = useRef(options);
 
   const handleReverification = useMemo(() => {
     const handler = createReverificationHandler({
       onOpenModal: __experimental_openUserVerification,
+      ...options,
     })(fetcherRef.current);
     return [handler] as const;
-  }, [__experimental_openUserVerification, fetcherRef.current]);
+  }, [__experimental_openUserVerification, fetcherRef.current, optionsRef.current]);
```

Comment:
I don't think it's necessary to put ref values in the dependency array, as they are mutable outside of the react lifecycle. When they change they wouldn't trigger a re-execution of the memoized value here.